features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/sha/sha1_alt_sw.c

```diff
@@ -24,17 +24,10 @@
  *  http://www.itl.nist.gov/fipspubs/fip180-1.htm
  */
 
-#if !defined(MBEDTLS_CONFIG_FILE)
-#include "mbedtls/config.h"
-#else
-#include MBEDTLS_CONFIG_FILE
-#endif
+#include "mbedtls/sha1.h"
 
 #if defined(MBEDTLS_SHA1_C)
 #if defined(MBEDTLS_SHA1_ALT)
-
-#include "mbedtls/sha1.h"
-
 #include <string.h>
 #if defined(MBEDTLS_SELF_TEST)
```

Comment:
I think this file does not have a self test, so this code seems slightly redundant.

Reply:
@andresag01 `sha1_alt_sw.c` is a duplicate of `sha1.c` and works as fallback for H/W. I just make needed modifications and leave others unchanged. The redundancy is ignored.

Reply:
@ccli8: Could you please help me understand why is there a software fallback for SHA1 at all? I mean, it seems that there is an accelerator available. Is it because for some reason it is not possible to switch the accelerator context so that operations can be interleaved?

P.S. The same comment does for SHA256 and SHA512

Reply:
@andresag01 Yes, you got it. The SHA accelerator doesn't support context save & restore and can only operate to the end without interleaving. So as there is context clone (`mbedtls_sha1_clone()`) or a new context request (`mbedtls_sha1_init()`), it can only resort to S/W fallback. Besides, SHA1/SHA256/SHA512 share the same accelerator.